src/client/modals/url-shortener-modal/url-shortener-modal.tsx

```diff
@@ -60,6 +60,12 @@ export class UrlShortenerPrompt extends React.Component<UrlProp, UrlShortenerPro
 
   shortenUrl() {
     return fetch("shorten?url=" + encodeURIComponent(this.props.url))
+      .then(response => {
+        if (!response.ok) {
+          throw Error(response.statusText);
```

Comment:
How it will be presented on UI, do you have any screenshot?
Does it make sense to write unit test with mock?